drivers/gpio/gpio_mchp_xec_v2.c

```diff
@@ -81,103 +82,118 @@ static inline void xec_mask_write32(uintptr_t addr, uint32_t mask, uint32_t val)
 	sys_write32(r, addr);
 }
 
-/*
- * notes: The GPIO parallel output bits are read-only until the
- * Alternate-Output-Disable (AOD) bit is set in the pin's control
- * register. To preload a parallel output value to prevent certain
- * classes of glitching for output pins we must:
- * Set GPIO control AOD=1 with the pin direction set to input.
- * Program the new pin value in the respective GPIO parallel output
- * register.
- * Program other GPIO control bits except direction.
- * Last step set the GPIO control register direction bit to output.
+/* NOTE: gpio_flags_t b[0:15] are defined in the dt-binding gpio header.
+ * b[31:16] are defined in the driver gpio header.
+ * Hardware only supports push-pull or open-drain.
  */
-static int gpio_xec_configure(const struct device *dev,
-			      gpio_pin_t pin, gpio_flags_t flags)
+static int gpio_xec_validate_flags(gpio_flags_t flags)
 {
-	const struct gpio_xec_config *config = dev->config;
-	uintptr_t pcr1_addr = pin_ctrl_addr(dev, pin);
-	uintptr_t pout_addr = pin_parout_addr(dev);
-	uint32_t pcr1 = 0U;
-	uint32_t mask = 0U;
-
-	/* Validate pin number range in terms of current port */
-	if ((valid_ctrl_masks[config->port_num] & BIT(pin)) == 0U) {
-		return -EINVAL;
+	if ((flags & (GPIO_SINGLE_ENDED | GPIO_LINE_OPEN_DRAIN))
+	    == (GPIO_SINGLE_ENDED | GPIO_LINE_OPEN_SOURCE)) {
+		return -ENOTSUP;
 	}
 
-	/* Don't support "open source" mode */
-	if (((flags & GPIO_SINGLE_ENDED) != 0U) &&
-	    ((flags & GPIO_LINE_OPEN_DRAIN) == 0U)) {
-		return -ENOTSUP;
+	if ((flags & GPIO_OUTPUT_INIT_LOW) && (flags & GPIO_OUTPUT_INIT_HIGH)) {
+		return -EINVAL;
 	}
 
-	/* The flags contain options that require touching registers in the
-	 * PCRs for a given GPIO. There are no GPIO modules in Microchip SOCs!
-	 * Keep direction as input until last.
-	 * Clear input pad disable allowing input pad to operate.
-	 * Clear Power gate to allow pads to operate.
-	 */
-	mask |= MCHP_GPIO_CTRL_DIR_MASK;
-	mask |= MCHP_GPIO_CTRL_INPAD_DIS_MASK;
-	mask |= MCHP_GPIO_CTRL_PWRG_MASK;
-	pcr1 |= MCHP_GPIO_CTRL_DIR_INPUT;
+	return 0;
+}
 
-	/* Figure out the pullup/pulldown configuration and keep it in the
-	 * pcr1 variable
-	 */
-	mask |= MCHP_GPIO_CTRL_PUD_MASK;
-
-	if ((flags & GPIO_PULL_UP) != 0U) {
-		/* Enable the pull and select the pullup resistor. */
-		pcr1 |= MCHP_GPIO_CTRL_PUD_PU;
-	} else if ((flags & GPIO_PULL_DOWN) != 0U) {
-		/* Enable the pull and select the pulldown resistor */
-		pcr1 |= MCHP_GPIO_CTRL_PUD_PD;
+/* Each GPIO pin has two 32-bit control registers. Control 1 configures pin
```

Comment:
Conventionally, block comments in Zephyr usually keep `/*` and `*/` on a separate lines.

https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style
https://kernel.org/doc/html/latest/process/coding-style.html#commenting

Reply:
Updated the block comments as per coding-style.